transcom/mymove
# B-21583 - update code to calculate gbloc for AK

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1025118)

## [Previous INT PR](https://github.com/transcom/mymove/pull/14451)
## [Revert of above INT PR](https://github.com/transcom/mymove/pull/14508)
## [INT PR again](https://github.com/transcom/mymove/pull/14581)

## Summary

When a move is created that has an Alaska duty location, we need to make it so it routes to the right GBLOC for that zone.
Needs to move through SC, TOO, TIO and show in the correct GBLOC for each queue.

### How to test

1. Login as a customer, create a new move with Origin AK city for each Zone and scenario below.  Six moves total.  Zone tips are below.  
There is some sql below too to help make sure your city/zip is pulling back the correct GBLOC you expect.
*See the attached spreadsheet of AK cities in the Agility story.
2. Be sure to use the same AK city for origin Duty Location and AK Pickup Address to make things easier.
3. Add HHG shipment, Submit move as customer.
4. Create an SC for each GBLOC (MBFL, JEAT, MAPS, MAPK)to verify that the moves show in the queue as expected.
5. As the SC, do a search under Move Search tab, make sure the Origin GBLOC shows as expected in the results.
6. Before submitting as SC, also login as the HQ user and verify you see the moves in the queue there as well.
7. Submit your moves and then login as TOO for each move and verify you see them in the queue for the pickup address you chose.
8. Approve your moves, login as Prime Sim, Update Shipment (adding pickup dates and weights).
9. DO NOT add any new service items, that work is done in another BL.  You will only be creating a Payment Request so that the move can go to the TIO.  There's a little bit of work to get the Payment request to work for INTL at this point.  On the main move details page as prime, scroll down to the bottom and click the Edit button next to the PODFSC.  Add a valid Port code (you can use SEA) and save.  Now you can go do the Payment Request, you can only do the Fuel surcharge at this point. 
10. Login as TIO users and verify you can see all moves in the queues as expected.
11.  Now go back through the steps above and create orders that have the destination of AK and each of the 6 zones.


**Zones**:
- Zone I shipments will be processed by JPPSO Elmendorf-Richardson (GBLOC MBFL).
- Zone II shipments are divided between JPPSO-NW (GBLOC JEAT) and JPPSO Elmendorf-Richardson (GBLOC MBFL) in the following manner:   JPPSO-NW (GBLOC JEAT) will process all shipments that are aligned with Zone II excluding Air Force and Space Force personnel.  Air Force and Space Force personnel in Zone II will be processed by JPPSO Elmendorf-Richardson (GBLOC MBFL).
- Zone III shipments will be processed by USCG Base Kodiak (GBLOC MAPS).
- Zone IV shipments will be processed by USCG Base Ketchikan (GBLOC MAPK) with the exception of Cordova which will be processed by USCG Base Kodiak (GBLOC MAPS).

**SQL** to run for zone/gbloc on addresses:
select j.code gbloc, j.name jppso, r.code rate_area, r.name rate_area_name, g.department_indicator, g.shipment_type 
from addresses a, v_locations v, re_oconus_rate_areas o, re_rate_areas r, jppso_regions j, gbloc_aors g 
where a.id = 'put_your_addr_id_here' 
and a.us_post_region_cities_id = v.uprc_id 
and v.uprc_id = o.us_post_region_cities_id 
and o.rate_area_id = r.id 
and o.id = g.oconus_rate_area_id 
and j.id = g.jppso_regions_id;
